src/libsyntax/ast.rs

```diff
@@ -1135,26 +1134,17 @@ pub enum ExprKind {
     Cast(P<Expr>, P<Ty>),
     /// A type ascription (e.g., `42: usize`).
     Type(P<Expr>, P<Ty>),
+    /// A `let pat = expr` pseudo-expression that only occurs in the scrutinee
+    /// of `if` / `while` expressions. (e.g., `if let 0 = x { .. }`).
+    Let(Vec<P<Pat>>, P<Expr>),
```

Comment:
why is this a `Vec`? For giving multiple patterns with `|`? Please document this here.

Reply:
Correct; We permit `if let p | q = ...` today. Updated the docs in https://github.com/rust-lang/rust/pull/60861/commits/ccb40bce46c10855927f456e607aa553336bc1c2.